README.md

```diff
@@ -32,3 +32,8 @@ The latest source code can be checked out from Github via:
 And then you can do a build by invoking `./gradlew build` from the top-level directory. You should also be able to run test suite by running `./gradlew test`
 
 If you are interested in this project, either as a user or as a developer, you may [write me]("mailto:revusky@NOSPAMjavacc.com"). Better yet, you can sign up on our[Discourse forum](https://discuss.parsers.org/) and post any questions or suggestions there.
+
+## License
+
+JavaCC21 is licensed under [BSD-3-Clause license](LICENSE).
+The code that is generated by JavaCC21 can be used under [CC0 (public domain) license](https://creativecommons.org/share-your-work/public-domain/cc0/).
```

Comment:
Well, again, this is simply not correct. The code that is generated by JavaCC21 can be used AS YOU SEE FIT. IT IS YOURS. This is what needs to be said, NOT that it is in the public domain.

Reply:
> AS YOU SEE FIT. IT IS YOURS

What is the well-known lawyer-approved license name for that?

Reply:
Vladimir, as regards this statement that the code you generate with the tool "IS YOURS TO DO WITH AS YOU SEE FIT" I would have to ask you a couple of very direct, pointed questions here:

1. Do you yourself not understand what this sentence means? 
2. Is there anybody you know (in this programming environment, obviously) who would not understand what this sentence means?
3. Is there any _reasonable_ doubt about the meaning of the sentence such that somebody would feel the need to consult a lawyer?

In the JavaCC FAQ that was (for a long time) maintained by some university professor in Canada, one of the FAQ's is: [1.8  What legal restrictions are there on JavaCC?
](http://www.engr.mun.ca/~theo/JavaCC-FAQ/javacc-faq-moz.htm#tth_sEc1.7) and you see there Professor Norvell's answer:

> There are essentially no restrictions on the use of JavaCC. In particular you may use the Java files that JavaCC produces in any way, including incorporating them into a product that you sell. JavaCC is freely redistributable under the its open source licence.  See the licence for details.

Now, I recognize that the FAQ is not any official statement by anybody. However, I do not believe that the legacy JavaCC project ever made any statement more precise than the one above. More or less: _the code that you generate with the tool is yours and you can do with it what you want_. 

So, basically, unless the answer to any of the above questions is not what I think it is, I'm really just going to have to close this PR as frivolous. You have to understand my position. I am at a juncture where I am desperate to find competent, energetic collaborators. And I was hoping that you would be one. And I still have that hope, but we're getting to a situation where you seem obsessed with fixing things that are not really broken. The existing build system was not broken. It worked pretty well and there was really no urgent need to do anything with it. Except now it is broken!!! Because the tests aren't working. Since you started this process of migrating to Gradle, I really think you ought to finish that before starting to nitpick about other things.

JavaCC 21 is under the exact same license as the legacy tool and there is a 23 year history at this point in which nobody has ever had any real questions about their ability to use JavaCC for whatever they want. I checked and the jarfiles of the legacy JavaCC do not have a license file in them. Nor do the ANTLR jarfiles. I mean, the ones you download via Maven, like here: 

https://mvnrepository.com/artifact/net.java.dev.javacc/javacc

I completely agree that this project should also have its jarfiles up there. So, yes, I would appreciate some concrete help in making this happen. But I don't want to argue about ersatz licensing issues. JavaCC 21 is under the same license as legacy JavaCC. The source code can be used and modified under the BSD license and any generated code is just yours. There is no problem or ambiguity with this.

Reply:
@revusky , it is really sad you make judgments in the field you have no expertise in.

The ambiguity is there as long as a non-standard license is used.
You might treat https://spdx.org/licenses/ as a definition of a standard.
Note: there's no "code is just yours" license there.

--

> 3. Is there any reasonable doubt about the meaning of the sentence such that somebody would feel the need to consult a lawyer?

I'm not a lawyer, and I was told that in 100% of the cases I must ask a lawyer if I need licensing advice. The exception is the case when exactly the same question has already been resolved.

So, the answer to question 3 is "yes, everybody must consult their lawyer before using javacc21".

---

> legacy JavaCC project ever made any statement more precise

You were going to make a better JavaCC, weren't you?
A moot license is one of the issues of the legacy.
I see you are keen on making code more readable, and making licensing clear is very important as well for the users.

---

>The existing build system was not broken

It was broken (just from the top of my head):
* Project did not load to IDE
* Dependencies from Central could not be used (e.g. testing frameworks)
* Code style was not checked at all (e.g. different files were committed with different ends-of-lines, lots of files had trailing whitespace, and there were cases when tabs and spaces were mixed)
* Publishing to Central was not there

All of the above is fixed now.
**If you consider it is broken now, feel free to revert the whole thing.**

---

> I really think you ought to finish that before starting to nitpick about other things.

Ok, go ahead suggest to me what I should finish.
Apparently, licensing is the very next step that should be adjusted before the artifact could be released to Central.

Just in case: I did license management for Apache JMeter which has **92** third-party dependencies. Luckily, most of them did embed license files to the jars, so I did not have to manually inspect project websites.
In every case the license was missing I created an issue to include the license. You can search for `Dependency with "manual" license configuration` [here](https://github.com/apache/jmeter/pull/469)

Reply:
Look, this is just getting exruciating. JavaCC21 is released under the same terms as the legacy JavaCC tool. So, if people really needed to consult a lawyer to use one, they would need to consult a lawyer to use the other one. How many of the existing users of JavaCC have consulted a lawyer about using it? I have no idea, but surely not many. It is simply very well understood that you can download the thing and use it. It's just not an issue for anybody.

Now, I have no objection to putting a license file into a .jar file (why would I? Who cares?) but I did make the simple point that neither JavaCC nor ANTLR do this. Of course, it's easy to find the license on the respective websites. 

Again, JavaCC21 has the same license as the legacy JavaCC. There is only so much mental energy one can put into something that is obviously just not a problem.